### PR TITLE
Show more

### PR DESCRIPTION
**Pull request by minhphan156**
# Description

Include a summary of the change and which ticket has been completed.

#### Completes ticket: <Issue #>

# Tests

Describe the tests that you ran to verify your changes.

---

# Reviewer Checklist

- [ ] I have reviewed the changes.
- [ ] The code follows good style.
- [ ] I have reviewed the logic of the code.
- [ ] I have performed the same tests locally listed above, and receieved the same/similar results.

---

_By submitting this Pull Request, you hereby acknowledge and agree that you are following the Pull Request rules found on the General wiki._

### CODE REVIEW

File: src/components/post/Post.js [hunk 1]
```diff
@@ -123,7 +124,7 @@ class Post extends Component {
             <Grid container justify="space-between" alignItems="flex-start">
               <Grid className="KnowledgePostText" item>
                 {upload.metadata.authors}
-                <div className="KnowledgePostTime">{upload.metadata.time}</div>
+                <div className="KnowledgePostTime">{upload.metadata.time || upload.metadata.createdAt}</div>
```
Comment (SandroSal) [inline src/components/post/Post.js]: what's the difference between time and createdAt?

File: src/components/post/Post.js [hunk 2]
```diff
@@ -52,32 +52,33 @@ const styles = theme => ({
...
-      // time: "",
-      // tags: [],
-      // summary: "",
-      // dataset: null,
-      // notebook: null,
-      // isSubmit: false,
-      // views: 167
-    };
-  }
+  // constructor() {
```
Comment (SandroSal) [inline src/components/post/Post.js]: if unneeded, can we delete it ?

File: src/components/post/Post.js [hunk 3]
```diff
@@ -52,32 +52,33 @@ const styles = theme => ({
...
+  //     // summary: "",
+  //     // dataset: null,
+  //     // notebook: null,
+  //     // isSubmit: false,
+  //     // views: 167
+  //   };
+  // }
 
   render() {
+    console.log("POST VIEW RENDERING")
```
Comment (SandroSal) [inline src/components/post/Post.js]: needed console log?

File: src/components/post/Post.js [hunk 4]
```diff
@@ -109,9 +110,9 @@ class Post extends Component {
                   alignItems="flex-start"
                 >
                   <Grid className="viewContainer" className={classes.viewCount}>
-                    {upload.metadata.views == null
-                      ? "0"
-                      : upload.metadata.views}
+                    {upload
```
Comment (SandroSal) [inline src/components/post/Post.js:98]: as far as I know, views is not returned after an upload ... kinda curious how your testing went on this one

File: src/components/post/Post.js [hunk 5]
```diff
@@ -140,7 +141,7 @@ class Post extends Component {
                 <a
                   target="_blank"
                   rel="noopener noreferrer"
-                  href={upload.metadata.dataset}
+                  href={upload.metadata.dataset ? upload.metadata.dataset : ""}
```
Comment (SandroSal) [inline src/components/post/Post.js]: don't think this is needed, can we go with the old version (on the left in red?)

File: src/components/post/Post.js [hunk 6]
```diff
@@ -171,15 +174,17 @@ class Post extends Component {
             <Grid>
               <div
                 id="notebook-container"
-                dangerouslySetInnerHTML={{ __html: upload.html.final_html }}
+                dangerouslySetInnerHTML={{ __html: upload.html.final_html || upload.html}}
```
Comment (SandroSal) [inline src/components/post/Post.js:162]: what's the or statement for?

File: src/components/post/Post.js [hunk 7]
```diff
@@ -171,15 +174,17 @@ class Post extends Component {
...
             </Grid>
           </Paper>
           <br></br>
           <br></br>
         </Grid>
       );
-    } else {
+    } 
+    else {
+      console.log('return null post')
```
Comment (SandroSal) [inline src/components/post/Post.js:172]: console log, please remove

File: src/components/explore/ExploreCard.js [hunk 8]
```diff
@@ -128,6 +132,16 @@ const styles = {
...
 class ExploreCard extends Component {
+  handleShowMore = () => {
+    this.props.auth
+    .getAccessToken()
+    .then(token => {
+      this.props.getIndividualPostsActions(this.props.post, token, this.props.history);
+    })
+    .catch(err => console.log(err));
+
+    // this.props.history.push("/post")
```
Comment (SandroSal) [inline src/components/explore/ExploreCard.js]: delete please

File: src/actions/getPostsActions.js [hunk 9]
```diff
@@ -98,6 +99,38 @@ export const getFilteredPostsActions = (
...
+export const getIndividualPostsActions = (
+  post,
+  accessToken,
+  history
+) => dispatch => {
+ 
+  let config = {
+    headers: { Authorization: `Bearer ${accessToken}` },
+  };
+  console.log("getIndividualPostsActions-", post);
```
Comment (SandroSal) [inline src/actions/getPostsActions.js]: console log

**Review by SandroSal — CHANGES_REQUESTED** [review]

File: src/components/post/Post.js (hunk 4 above)
Comment (SandroSal) [reply src/components/post/Post.js:98]: just tested it... it doesn't work

File: src/components/post/Post.js (hunk 1 above)
Comment (minhphan156) [reply src/components/post/Post.js]: the endpoint to retrieve post when you click **showmore** only return createdAt

File: src/components/post/Post.js (hunk 4 above)
Comment (minhphan156) [reply src/components/post/Post.js:98]: then it will return 0 for upload

File: src/components/post/Post.js (hunk 6 above)
Comment (minhphan156) [reply src/components/post/Post.js:162]: the endpoint to retrieve post when you click showmore doesn't have **final_html prop**

File: src/components/post/Post.js [hunk 10]
```diff
@@ -52,32 +52,33 @@ const styles = theme => ({
...
+
     let dataLinkButton;
     let dataLink;
-
     if (upload != null) {
       dataLink = "/" + upload.metadata.dataset;
 
       // if there is a dataset, display a link to it
-      if (upload.metadata.dataset != "") {
+      if (upload.metadata.dataset !== "") {
```
Comment (SandroSal) [inline src/components/post/Post.js]: for your implementation, if there is no dataset you don't get anything returned, so you would have to enhance this if statement, something like this:

|| upload.metadata.dataset === null

File: src/components/post/Post.js (hunk 4 above)
Comment (SandroSal) [reply src/components/post/Post.js:98]: for it to work, please change it back to the way it was

File: src/components/post/Post.js (hunk 10 above)
Comment (minhphan156) [reply src/components/post/Post.js]: fixed

File: src/actions/getPostsActions.js (hunk 9 above)
Comment (minhphan156) [reply src/actions/getPostsActions.js]: removed

File: src/components/explore/ExploreCard.js (hunk 8 above)
Comment (minhphan156) [reply src/components/explore/ExploreCard.js]: done

File: src/components/post/Post.js (hunk 7 above)
Comment (minhphan156) [reply src/components/post/Post.js:172]: removed

File: src/components/post/Post.js (hunk 5 above)
Comment (minhphan156) [reply src/components/post/Post.js]: fixed

File: src/components/post/Post.js (hunk 2 above)
Comment (minhphan156) [reply src/components/post/Post.js]: removed

File: src/components/post/Post.js (hunk 3 above)
Comment (minhphan156) [reply src/components/post/Post.js]: removed

**Pull request merged**